tasks/node_maintainance/node_health_monitor.py

```diff
@@ -0,0 +1,692 @@
+
+    try:
+        server_pool_helper = ServerPoolSDKHelper()
+        logger.info(f"Connection to Server Pool successful")
+    except Exception as e:
+        result["result"] = False
+        result["reason"] = f"Cannot connect to Server Pool using SDK : {e}"
+        logger.error(result["reason"])
+        return result
+
+    try:
+        remote_connection_helper = RemoteConnectionObjectFactory.fetch_helper(ipaddr,"root","couchbase")
+    except Exception as e:
+        result["result"] = False
+        result["reason"] = f"The node {ipaddr} is unreachable, cannot perform os version checks : {e}"
+        logger.error(result["reason"])
+        return result
+
+    try:
+        memory_in_node = remote_connection_helper.find_memory_total()
+    except Exception as e:
+        result["result"] = False
+        result["reason"] = f"Could not find total memory for node {ipaddr} : {e}"
+        logger.error(result["reason"])
+        return result
+
+    if "memory" in doc and memory_in_node == doc["memory"]:
+        doc["tags"]["memory_node_check"] = {
+            "memory_node_match" : True
+        }
+    else:
+        doc["tags"]["memory_node_check"] = {
+            "memory_node_match" : False,
+            "memory_in_node" : memory_in_node
+        }
+
+    try:
+        res = server_pool_helper.add_node_to_server_pool(doc)
+        if not res:
+            result["result"] = False
+            result["reason"] = f"Cannot upsert node {ipaddr} with node-memory-consistency checks to server pool"
+            logger.error(result["reason"])
+            return result
+        logger.info(f"Document for node {ipaddr} with node-memory-consistency checks upserted to server pool successfuly")
+    except Exception as e:
+        result["result"] = False
+        result["reason"] = f"Cannot upsert node {ipaddr} with node-memory-consistency checks to server pool : {e}"
+        logger.error(result["reason"])
+        return result
+
+    result["result"] = True
+    result["os_node_match"] = doc["tags"]["memory_node_check"]
+    return result
+
+def check_node_os_match(doc):
+    result = {}
+    ipaddr = doc["ipaddr"]
+
+    # TODO - Remove post cleaning up of server pool
+    if "tags" not in doc and "connection_check" not in doc["tags"]:
+        result["result"] = False
+        result["reason"] = f"OS version check was run before connection checks for {ipaddr}"
+        logger.error(result["reason"])
+        return result
+    elif not doc["tags"]["connection_check"]:
+        result["result"] = False
+        result["reason"] = f"The node is unreachable, cannot perform os version checks for {ipaddr}"
+        logger.error(result["reason"])
+        return result
+
+    try:
+        server_pool_helper = ServerPoolSDKHelper()
+        logger.info(f"Connection to Server Pool successful")
+    except Exception as e:
+        result["result"] = False
+        result["reason"] = f"Cannot connect to Server Pool using SDK : {e}"
```

Comment:
This can be moved to a function, taking result, reason and the exception as params

Reply:
Addressed in latest commit